model/task/task.go

```diff
@@ -1062,6 +1071,18 @@ func (t *Task) MarkUnscheduled() error {
 
 }
 
+func (t *Task) MarkUnattainableDependency(dependency *Task, unattainable bool) error {
```

Comment:
it seems like you're never going to want to set this to false... all dependencies are attainable, and once they become unattainable they never become attainable. 

also we should set the field value locally in the current object.